helpdesk-message-proxy-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/messageproxy/service/RoomService.java

```diff
@@ -38,17 +40,25 @@ public String newServiceStream(String ticketId, String groupId) {
     roomAttributes.setName("[" + groupId + "] Ticket Room #" + ticketId);
     roomAttributes.setReadOnly(false);
     roomAttributes.setPublic(false);
+    roomAttributes.setViewHistory(viewHistory);
 
-    Room room;
+    Room room = symphonyClient.getRoomService().createRoom(roomAttributes);
+    room.getRoomDetail().getRoomAttributes().setViewHistory(viewHistory);
+    LOGGER.info("Created new room: " + roomAttributes.getName());
 
-    try {
-      room = symphonyClient.getRoomService().createRoom(roomAttributes);
-      LOGGER.info("Created new room: " + roomAttributes.getName());
+    return room;
+  }
 
-      return room.getStreamId();
+  public Room createServiceStream(String ticketId, String groupId) {
+    try {
+      return newServiceStream(ticketId, groupId, Boolean.TRUE);
     } catch (RoomException e) {
-      LOGGER.error("Create room failed: ", e);
-      return null;
+      try {
+        return newServiceStream(ticketId, groupId, Boolean.FALSE);
+      } catch (RoomException e1) {
+        LOGGER.error("Create room failed: ", e1);
+        return null;
```

Comment:
You cannot return a null value in this method. You shouldn't create a new ticket if you don't have success to create 
the service stream.

You must throw this exception and the caller should send an error message to the client stream informing you had issues to create the ticket.